File: python/7 kyu/408.py
# https://www.codewars.com/kata/588453ea56daa4af920000ca/train/python

# Passed

def array_packing(arr):
    binary_nums = [bin(num).split("b")[1].zfill(8) for num in arr[::-1]]
    result_binary = "".join(binary_nums)
    result = int(result_binary, 2)
    return result

output = array_packing([23, 45, 39])
print(output)

File: nodejs/7 kyu/408.js
// https://www.codewars.com/kata/59de469cfc3c492da80000c5/train/javascript

// Passed

function compress(sentence) {
  let wordToValueMap = new Object();
  let words = sentence.toLowerCase().split(" ");
  let currentVal = 0;
  let result = "";
  for (const word of words) {
    if (!(word in wordToValueMap)) {
      wordToValueMap[word] = currentVal;
      currentVal++;
    }

    let wordVal = wordToValueMap[word];
    result += String(wordVal);
  }

  return result;
}

const output = compress("SILLY LITTLE BOYS silly little boys");
console.log(output);